include/mata/nfa.hh

```diff
@@ -958,6 +958,19 @@ Nfa uni(const Nfa &lhs, const Nfa &rhs);
 Nfa intersection(const Nfa& lhs, const Nfa& rhs,
                  bool preserve_epsilon = false, std::unordered_map<std::pair<State, State>, State> *prod_map = nullptr);
 
+/**
+ * @brief Compute intersection of two NFAs with a possibility of using multiple epsilons.
+ *
+ * @param[in] lhs First NFA to compute intersection for.
+ * @param[in] rhs Second NFA to compute intersection for.
+ * @param[in] preserve_epsilon Whether to compute intersection preserving epsilon transitions.
+ * @param[in] epsilons Set of symbols to be considered as epsilons
+ * @param[out] prod_map Mapping of pairs of the original states (lhs_state, rhs_state) to new product states.
+ * @return NFA as a product of NFAs @p lhs and @p rhs with ε-transitions preserved.
+ */
+Nfa intersection_eps(const Nfa& lhs, const Nfa& rhs, bool preserve_epsilon, const std::set<Symbol>& epsilons,
```

Comment:
I am not sure whether we want this here or we should move it to `strings` or `algorithms`

Reply:
I don't have a strong opinion to that. If the others are not strictly against, I switch it to`algorithms` header file.

Reply:
I moved the functions to `Algorithm` namespace